caddyconfig/caddyfile/lexer.go

```diff
@@ -168,3 +169,23 @@ func (l *lexer) next() bool {
 		val = append(val, ch)
 	}
 }
+
+// Tokenize takes bytes as input and lexes it into
+// a list of tokens that can be parsed as a Caddyfile.
+// Also takes a filename to fill the token's File as
+// the source of the tokens, which is important to
+// determine relative paths for `import` directives.
+func Tokenize(input []byte, filename string) ([]Token, error) {
```

Comment:
Any good reason we need to change from `string` to `[]byte` here? Now all the tests have to change...

Reply:
Because parser.go uses a []byte instead. I tried the other way but converting the []byte back to string doesn't seem like a good idea.

Reply:
Fair enough